### DISCUSSION TITLE
Welcome to synology-api Discussions!

### DISCUSSION THREAD

**Discussion by N4S4** [N4S4/synology-api#89] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of the community. You can use it for:
  * Ask questions you’re wondering about.
  * Solve a issue.
  * Share ideas.
  * Make a request.
  * Engage with other community members.
  
Feel free to ask anything regarding the package and remember that this is a community we build together 💪 so be kind and respectful to others!

**Comment by roymoran** [comment]
Is it possible to use this API to make modifications to settings in the control panel? 

I would like to automate things like creating new rules for the reverse proxy, generate SSL certificates, associate it to a service, as well as add DDNS rules.

**Reply by N4S4** [reply]
Atm limited functions are able to modify control panel settings, you can navigate through file naming, for example core_certificate.py, or you can check a list of available functions in the Readme file.
Hopefully I will include more functions soon.

**Comment by darth-anish** [comment]
Exception raised when trying to instantiate FileStation object in localhost.

Exception: **requests.exceptions.ConnectionError: HTTPConnectionPool(host='https', port=80)**

Any idea how to resolve this?

**Reply by N4S4** [reply]
Could you please show me the entire log? Including your input

**Reply by darth-anish** [reply]
Input:

fs = filestation.FileStation('https://login.aqysta.com','443',Username,password,secure=False,cert_verify=False,dsm_version=7,debug=True,otp_code=None)

Output Error:

raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='https', port=80): Max retries exceeded with url: //login.aqysta.com:443/webapi/auth.cgi?api=SYNO.API.Auth&version=7&method=login&account=Username&passwd=password&session=FileStation&format=cookie (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f30e83a9e20>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))

**Reply by N4S4** [reply]
The port number must be an integer you must remove the quotes from it, also as you are using an https connection `secure=True` and `cert_verify=True` (if you are using a certificare but I guess so)

**Reply by darth-anish** [reply]
Input:

fs = filestation.FileStation('https://login.aqysta.com',443,Username,pw,secure=True,cert_verify=True,dsm_version=7,debug=True,otp_code=None)


Error message:

raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='https', port=80): Max retries exceeded with url: //login.aqysta.com:443/webapi/auth.cgi?api=SYNO.API.Auth&version=7&method=login&account=Username&passwd=pw&session=FileStation&format=cookie (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fde07bc5460>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))

**Reply by darth-anish** [reply]
@N4S4 I tried with new input and got new error message

Input

fs = filestation.FileStation('login.aqysta.com',443,Username,pw,secure=True,cert_verify=True,dsm_version=7,debug=True,otp_code=None](https://login.aqysta.com%27%2C443%2Cusername%2Cpw%2Csecure%3Dtrue%2Ccert_verify%3Dtrue%2Cdsm_version%3D7%2Cdebug%3Dtrue%2Cotp_code%3Dnone/))

Error message:

self._sid = session_request.json()['data']['sid']
KeyError: 'data'

**Reply by deep-mdlz** [reply]
@darth-anish Were you able to resolve this issue?

**Comment by MathieuSaintes** [comment]
Do you guys have documentation on each API available methods and parameters ?
Like, how did you know that the SYNO.ActiveBackup.Inventory API could/should be used with the list method, and that the SYNO.ActiveBackup.Overview API could/should be used with the list_device_transfer_size method and time_start, time_end parameters ?
I'm surprised not to find any documentation from google !

**Reply by N4S4** [reply]
Hi, 

The only API documentation available is the one from synology dev website, the remaining has been discovered by sniffing requests to the NAS

**Comment by deep-mdlz** [comment]
Hi @N4S4 

I am getting the following error while hiting the filestation API. 

(MaxRetryError("HTTPConnectionPool(host='nas-server-it.sg4.quickconnect.to', port=5000): Max retries exceeded with url: /webapi/auth.cgi?api=SYNO.API.Auth&version=7&method=login&account=USERNAME&passwd=PASSWORD&session=FileStation&format=cookie (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x0000023377344070>, 'Connection to nas-server-it.sg4.quickconnect.to timed out. (connect timeout=None)'))"),)

**Reply by N4S4** [reply]
Unfortunately it does not work with quickconnect, you should set a domain to make it work

**Reply by deep-mdlz** [reply]
@N4S4 Can you please help me how i can set a domain

**Reply by N4S4** [reply]
This two links might help you

[Certificate](https://mariushosting.com/synology-how-to-enable-https-on-dsm-7/)

[hostname](https://mariushosting.com/synology-how-to-add-wildcard-certificate)

**Comment by abraxas48** [comment]
Hi there,
I'm using the Download Station API to automatically retrieve statistics on a daily basis from the NAS. The issue I'm encountering is that when I don't open DSM during the day, the information collection is not accurate. It seems like I need to open DSM for it to update the Download Station statistics and ensure accurate data collection.
Has anyone else experienced this issue? Any solutions?

**Reply by N4S4** [reply]
Hi, I have to say i never run a data collection for long time. By not accurate you mean freezes at some value?

**Comment by lisa-rojas** [comment]
Where did you get the API reference for the SYNO.Backup endpoints, I need to work with this but haven't found the API reference for it on the Synology official site...
It would be awesome if you could help me with that!!

Pura Vida!!
Thanks!!

**Reply by N4S4** [reply]
Hi Lisa,

Most of this wrapper end-points have been sniffed via browsers while using web DSM or with similar way. You may try with browser developer tool while making requests

**Comment by ghost** [comment]
Hi!  I wonder if you plan to create an asynchronous version using aiohttp for example.

**Reply by N4S4** [reply]
Hello, no plan for that atm
